src/middlewared/middlewared/plugins/smb.py

```diff
@@ -598,6 +606,7 @@ async def validate_smb(self, new, verrors):
         Str('dirmask'),
         Bool('ntlmv1_auth'),
         Bool('multichannel', default=False),
+        Str('encryption', enum=['DEFAULT', 'NEGOTIATE', 'DESIRED', 'REQUIRED']),
```

Comment:
So, this parameter **_must_** be supplied?  (i.e. no default of DEFAULT?)

Reply:
No. If it is omitted then current value is preserved. I left it as this to prevent accidental changes to transport encryption settings via incomplete payload.